client/blocks/comment-button/index.jsx

```diff
@@ -42,19 +42,28 @@ const CommentButton = React.createClass( {
 			commentCount = this.props.commentCount;
 
 		if ( commentCount === 0 ) {
-			label = this.translate( 'Comment' );
+			label = <span className="comment-button__label-status">
+				{ this.translate( 'Comment', { context: 'verb' } ) }
+			</span>;
 		} else {
 			label = this.translate(
-				'Comment',
-				'Comments', {
+				'{{count/}}{{span}}Comment{{/span}}',
+				'{{count/}}{{span}}Comments{{/span}}', {
+					components: {
+						count: commentCount > 0 ? ( <span className="comment-button__label-count">
+							{ commentCount }
+						</span> ) : null,
+						span: <span className="comment-button__label-status" />
+					},
 					count: commentCount,
 				}
 			);
 		}
 
 		const labelElement = ( <span className="comment-button__label">
-			{ commentCount > 0 ? <span className="comment-button__label-count">{ commentCount }</span> : null }
-			{ this.props.showLabel && <span className="comment-button__label-status">{ label }</span> }
+			{ this.props.showLabel ? label : commentCount > 0 && <span className="comment-button__label-count">
```

Comment:
If `showLabel` then show the label (with the new number component). Otherwise, if the comment count is `> 0` display just the count.